silkworm/rpc/core/evm_executor.cpp

```diff
@@ -240,8 +240,8 @@ ExecutionResult EVMExecutor::call(
 
     const auto owned_funds = execution_processor_.intra_block_state().get_balance(*txn.sender());
 
-    if (const auto result = protocol::validate_call_funds(txn, evm, owned_funds, bailout);
-        result != ValidationResult::kOk) {
+    if (const auto result = protocol::validate_call_funds(txn, evm, owned_funds);
+        !bailout && result != ValidationResult::kOk) {
```

Comment:
The application is a big awkward because we call the function even if we don't need the result. But considering the bailout is not enabled often this somehow guarantees the `validate_call_funds` still work correctly without crashes or assert failures.